project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -20,51 +27,58 @@ public DatabaseWrapper(String instanceId, String databaseId) {
   }
 
   /**
-   * Given a user, insert a row with all available fields into the DB
+   * Given a user, insert or update a row with all available fields into the DB
    *
-   * @param user the user to be inserted; user's ID field should not exist in DB
+   * @param user the user to be updated; user's email may or may not exist in DB
    */
-  public void insertUser(User user) {
+  public void insertOrUpdateUser(User user) {
+    // Given a user, update its corresponding row's new fields in DB
     SpannerOptions options = SpannerOptions.newBuilder().build();
```

Comment:
I am thinking maybe you can inject dbClient, so you don't need to create it in every function. 

You don't need to change this in the PR. Just a thought.

Reply:
Ack, might be an application-wide change if we have time, as we have many of these types of methods from different PRs.